src/main/java/net/emustudio/edigen/generation/GenerateMethodsVisitor.java

```diff
@@ -69,10 +74,22 @@ public void visit(Rule rule) throws SemanticException {
     @Override
     public void visit(Mask mask) throws SemanticException {
         boolean isZero = mask.getBits().containsOnly(false);
+        int maskStart = mask.getStart();
+        int maskLength = mask.getBits().getLength();
+        boolean alreadyRead = unitWasRead && unitLastStart == maskStart && unitLastLength == maskLength;
         
-        if (!isDefaultCase) {
-            put("unit = read(start + " + mask.getStart()
-                    + ", " + mask.getBits().getLength() + ");", true);
+        if (!isDefaultCase && !isZero && !alreadyRead) {
+            if (maskLength > 32) {
+                throw new SemanticException("Mask length can only be 32 bites (4 bytes)!", mask);
```

Comment:
Because unit size is 4 bytes

Reply:
Would this cause problems if someone wanted to implement a plugin for a 64-bit processor?

Reply:
Well - isn't there something like mask/pattern "splitting" pass which splits possibly long patterns by unit size? If not, we can   implement it, but it's not necessary since user can split his/her patterns to multiple subrules (but I didn't test it). However when I was looking at 64 bit x86 instructions, largest instruction has 15 bytes. But... tbh emuStudio emulating a 64 bit CPU? I strongly doubt it. When it will be the case, we can think of some solution then...

Just a side note, this change is assuming a fact that unit is of type `int`, which has 4 bytes only. If the mask/pattern was longer, it won't work even without this change, because data won't fit in the unit anyway.

Reply:
Actually `SplitVisitor` does it

Reply:
Then it is OK.